log/zerolog/factory.go

```diff
@@ -17,7 +17,7 @@ func Create(lvl log.Level) log.FactoryFunc {
 	zerolog.LevelFieldName = "lvl"
 	zerolog.MessageFieldName = "msg"
 	zerolog.TimeFieldFormat = time.RFC3339Nano
-	zl := zerolog.New(os.Stdout).With().Timestamp().Logger().Hook(sourceHook{skip: 7})
+	zl := zerolog.New(zerolog.SyncWriter(os.Stderr)).With().Timestamp().Logger().Hook(sourceHook{skip: 7})
```

Comment:
Nice catch. Will sync writer has a huge performace impact?
Is there any reason for using stderr vs stdout?

Reply:
I am also worried about that.. (the performance) and I could not find evidence of corrupted log lines, but perhaps some other system is filtering them out. If I cannot see them doesn't mean they're not there, considering that there are these big disclaimers about locking even in the Go `fmt` package source code. Go's `log` package does the sync writes from what I can see.

For `stdout`/`stderr` it's still arbitrary but the convention is to use `stderr` because you might give "parseable data" in the output. For example even Go's `log` package uses it: https://golang.org/src/log/log.go?s=9636:9664#L76

(GitHub Actions consider each line on stderr as a problem, but that is arbitrary on their side...)

Reply:
I found this while checking if we do synchronous writes or not for the logging, it appears we do. This PR should be completely unrelated to that but I thought to submit it anyways.

Reply:
Could we prove the perfromance impact with a benchmark?

Reply:
Well, first comes correctness and then performance; I am however re-reading zerolog and Go documentation to find whether writing to stdout/stderr is concurrency-safe or not.

There are different opinions on SO: https://stackoverflow.com/questions/30746166/is-os-files-write-threadsafe (see both answers)

I would like to find some definitive and authoritative source on whether it's safe so that we can avoid the `SyncWriter`.

Reply:
@mantzas it is guaranteed to be locked already by Go, see my other PR here: https://github.com/rs/zerolog/pull/235

So we can safely never use `SyncWriter` here. I will remove it from this PR, so that only the change about `Stderr` is left

Reply:
(We can also close the PR at this point; I still believe `Stderr` is more correct for logs but the important issue was about doing correct writes and that's proven now)

Reply:
I agree with stderr also! Let's keep it